zerver/lib/events.py

```diff
@@ -1957,6 +1957,9 @@ def _draft_update_action(i: int) -> None:
     elif event["type"] == "restart":
         # The Tornado process restarted.  This has no effect; we ignore it.
         pass
+    elif event["type"] == "push_device":
+        state["push_devices"][event["push_account_id"]]["status"] = event["status"]
+        state["push_devices"][event["push_account_id"]]["error_code"] = event.get("error_code")
```

Comment:
This seems weird; don't we need to look at `op` to know what to do? Suggests `test_events.py` is incomplete.